tidy3d/plugins/invdes/design.py

```diff
@@ -119,6 +122,54 @@ class InverseDesign(AbstractInverseDesign):
 
     _check_sim_pixel_size = check_pixel_size("simulation")
 
+    @pd.root_validator(pre=False)
```

Comment:
maybe could do this using the post init validators defined in Tidy3dBaseModel? would avoid needing to do values.get() stuff\

The basic idea is to overwrite this method and call your validation methods inside of it
https://github.com/flexcompute/tidy3d/blob/develop/tidy3d/components/base.py#L150

Reply:
sure. I was thinking it'd be better to validate as early as possible, and here this works, i.e. you don't need an instance of the class to check the validation. I can change it

Reply:
there are some other instance in this PR where that approach doesn't work and I use the post init validator

Reply:
ok, I'll leave it up to you, just letting you know so you know about it.